packages/core/src/languages/solidity/predeploys.ts

```diff
@@ -281,19 +257,71 @@ export const deploySphinxSystem = async (
         )
       ).wait()
     } else if (status === BigInt(1)) {
-      logger?.info(`[Sphinx]: Setting status for drip ${dripName}...`)
+      spinner?.start(`Setting status for drip ${dripName}...`)
       await (
         await Drippie.status(
           dripName,
           2,
           await getGasPriceOverrides(provider, owner, executionMode)
         )
       ).wait()
+      spinner?.succeed(`Finished setting status for drip ${dripName}`)
     } else {
       throw new Error(`Drip ${dripName} has archived status`)
     }
   }
-  logger?.info('[Sphinx]: finished creating relayer drips')
+  spinner?.succeed('Finished creating relayer drips')
+}
+
+export const deploySphinxSystem = async (
+  provider: SphinxJsonRpcProvider | HardhatEthersProvider,
+  signer: ethers.Signer,
+  relayers: string[],
+  executionMode: ExecutionMode,
+  includeManagedServiceRoles: boolean,
+  spinner?: ora.Ora
+): Promise<void> => {
+  const block = await provider.getBlock('latest')
+  if (!block) {
+    throw new Error('Failed to get latest block.')
+  }
+
+  for (const {
+    artifact,
+    constructorArgs,
+    expectedAddress,
+  } of getSphinxConstants()) {
+    const { abi, bytecode, contractName } = artifact
+
+    spinner?.start(`Deploying ${contractName}...`)
+
+    const contract = await doDeterministicDeploy(provider, executionMode, {
+      signer,
+      contract: {
+        abi,
+        bytecode,
+      },
+      args: constructorArgs,
+      salt: ethers.ZeroHash,
+    })
+
+    const addr = await contract.getAddress()
+    assert(addr === expectedAddress, `address mismatch for ${contractName}`)
+
+    spinner?.succeed(`Deployed ${contractName}, ${await contract.getAddress()}`)
+  }
+
+  spinner?.succeed(`Finished deploying Sphinx contracts`)
+
+  if (includeManagedServiceRoles) {
+    await assignManagedServiceRoles(
+      provider,
+      signer,
+      relayers,
+      executionMode,
+      spinner
+    )
+  }
```

Comment:
Previously, the `ensureSphinxAndGnosisSafeDeployed` function would fail on live networks since all the logic for configuring the managed service requires a specific signer. So, I went ahead and refactored this so that it only runs when we deploy the contracts ourselves.